src/sink.jl

```diff
@@ -1,67 +1,43 @@
-
-# TODO currently there is no appending or anything like that
-
-mutable struct Sink <: Data.Sink
-    filename::String
-    schema::Data.Schema
-    ctable::Metadata.CTable
-    io::IO
-    description::String
-    metadata::String
-    columns::Vector{ArrowVector}
-end
-
-function Sink(filename::AbstractString, sch::Data.Schema=Data.Schema(),
-              cols::AbstractVector{<:ArrowVector}=Vector{ArrowVector}(undef, size(sch,2));
-              description::AbstractString="", metadata::AbstractString="")
-    ctable = Metadata.CTable(description, 0, Metadata.Column[], FEATHER_VERSION, metadata)
-    io = open(filename, "w+")
-    Sink(filename, sch, ctable, io, description, metadata, cols)
-end
-function Sink(filename::AbstractString, df::DataFrame; description::AbstractString="",
-              metadata::AbstractString="")
-    Sink(filename, Data.schema(df), description=description, metadata=metadata)
-end
-
-# required by DataStreams
-function Sink(sch::Data.Schema, ::Type{Data.Column}, append::Bool, file::AbstractString;
-              reference::Vector{UInt8}=UInt8[], kwargs...)
-    Sink(file, sch)
-end
-function Sink(sink::Sink, sch::Data.Schema, ::Type{Data.Column}, append::Bool;
-              reference::Vector{UInt8}=UInt8[])
-    Sink(sink.filename, sch, sink.columns)
-end
-
-Data.streamtypes(::Type{Sink}) = [Data.Column]
-
-Base.size(sink::Sink) = size(sink.schema)
-Base.size(sink::Sink, i::Integer) = size(sink.schema, i)
-
-
 """
-    write(filename::AbstractString, df::DataFrame)
+    Feather.write(tbl, file::String; description="", metadata="")
 
-Write the dataframe `df` to the feather formatted file `filename`.
+Write any Tables.jl implementation as a feather-formatted file. Optionally, a `description`
+and `metadata` can be provided as Strings.
 """
-function write(filename::AbstractString, df::AbstractDataFrame)
-    sink = Feather.Sink(filename, df)
-    Data.stream!(df, sink)
-    Data.close!(sink)
-end
-
+function write end
 
-function Data.streamto!(sink::Sink, ::Type{Data.Column}, val::AbstractVector{T}, row, col) where T
-    sink.columns[col] = arrowformat(val)
+function write(file::String, df::DataFrame; kwargs...)
+    Base.depwarn("`Feather.write(file, df)` is deprecated in favor of `Feather.write(tbl, file`) where tbl can be any Tables.jl implementation", nothing)
+    return write(df, file)
 end
 
-# NOTE: the below is very inefficient, but we are forced to do it by the Feather format
-function Data.streamto!(sink::Sink, ::Type{Data.Column}, val::AbstractVector{Union{T,Missing}},
-                        row, col) where T
-    hasmissing = Compat.findfirst(ismissing, val)
-    sink.columns[col] = arrowformat(hasmissing == nothing ? convert(AbstractVector{T}, val) : val)
+write(filename::AbstractString; kwargs...) = x->write(x, filename; kwargs...)
+function write(tbl, filename::AbstractString; description::String="", metadata::String="")
```

Comment:
You realize of course that changing the argument order is a breaking change; perhaps one of the few breaking changes it's even posssible to make to this package since the interface is so minimal.  Is this no ordering some sort of global standard?  I'm open to changing it, but I think we should have a really good reason.  Last I checked, most other packages are `write(filename, obj)`.